File: WdtcCore/src/main/java/org/wdt/wdtc/auth/accounts/OfflineAccounts.java
package org.wdt.wdtc.auth.accounts;

import org.apache.log4j.Logger;
import org.wdt.wdtc.auth.BaseUser;
import org.wdt.wdtc.auth.User;
import org.wdt.wdtc.manger.FileManger;
import org.wdt.wdtc.utils.SkinUtils;
import org.wdt.wdtc.utils.StringUtils;
import org.wdt.wdtc.utils.WdtcLogger;
import org.wdt.wdtc.utils.gson.JSONUtils;

import java.io.IOException;
import java.util.UUID;

public class OfflineAccounts extends BaseUser {
    private static final Logger logmaker = WdtcLogger.getLogger(OfflineAccounts.class);
    private final String username;
    private final String UserUuid = StringUtils.cleanStrInString(UUID.randomUUID().toString(), "-");

    public OfflineAccounts(String username) {
        this.username = username;
    }


    @Override
    public User getUser() throws IOException {
        User user = new User();
        user.setUserName(username);
        user.setType(Accounts.AccountsType.Offline);
        user.setAccessToken("${auth_access_token}");
        user.setUuid(UserUuid);
        SkinUtils utils = new SkinUtils(getUtils().getSkinFile());
        utils.setUserSkinInput(OfflineAccounts.class.getResourceAsStream("/steve.png"));
        user.setHeadFile(utils.writeSkinHead());
        JSONUtils.writeObjectToJsonFile(FileManger.getUsersJson(), user);
        logmaker.info(user);
        return user;
    }

    public SkinUtils getUtils() {
        return new SkinUtils(username);
    }
}

File: core/core-impl/src/main/kotlin/org/wdt/wdtc/core/impl/auth/accounts/OfflineAccounts.kt
package org.wdt.wdtc.core.impl.auth.accounts

import org.wdt.wdtc.core.openapi.auth.Accounts.AccountsType
import org.wdt.wdtc.core.openapi.auth.LoginUser
import org.wdt.wdtc.core.openapi.auth.User
import org.wdt.wdtc.core.openapi.utils.*
import java.util.*

class OfflineAccounts(private val username: String) : LoginUser {
	
	private val headFile = ioAsync {
		SkinUtils(SkinUtils.getUserSkinFile(username)).apply {
			userSkinInput = getResourceAsStream("/assets/skin/steve.png")
		}.writeSkinHead()
	}
	private val uuid = ioAsync {
		UUID.randomUUID().toString().cleanStrInString("-")
	}
	
	override suspend fun getUser(): User = runOnIO {
		User(
			username, "\${auth_access_token}", AccountsType.OFFLINE, uuid.await(), headFile.await()
		)
	}
	
}
